build-tools/packages/build-tools/src/typeValidator/compatibility.ts

```diff
@@ -26,26 +26,49 @@ type requireAssignableTo<_A extends B, B> = true;
  * This will strip some type branding information which ideally would be kept for stricter checking, but without it, const enums show up as breaking when unchanged.
  */
 export const typeOnly = `
-// See 'build-tools/src/type-test-generator/compatibility.ts' for more information.
+// See 'build-tools/packages/build-tools/src/typeValidator/compatibility.ts' for more information.
+type ValueOf<T> = T[keyof T];
+type WellKnownSymbols = OnlySymbols<ValueOf<typeof Symbol>>;
+type OnlySymbols<T> = T extends symbol ? T : never;
+// Omit (replace with never) a key if it is a symbol thats not a well known symbol from global Symbol.
+type FilterKey<P> = symbol extends P
+	? P
+	: P extends symbol
+		? P extends WellKnownSymbols
+			? P
+			: never
+		: P;
 type TypeOnly<T> = T extends number
 	? number
 	: T extends string
-	? string
-	: T extends boolean | bigint | symbol
-	? T
-	: {
-			[P in keyof T]: TypeOnly<T[P]>;
-	  };
+		? string
+		: T extends boolean | bigint | symbol
+			? FilterKey<T>
+			: {
+					[P in keyof T as FilterKey<P>]: TypeOnly<T[P]>;
+				};
```

Comment:
I like the new indentation better, just checking that we confirmed prettier/biome won't complain about the change? I don't remember if the auto-generated files end up getting formatted or not.

Reply:
Our prettier ignore has `**/src/**/test/types/*.generated.ts` to cover this.